docs/edsl.md

```diff
@@ -0,0 +1,739 @@
+# C++ Tile EDSL
+
+The C++ Tile EDSL (Embedded Domain Specific Language) provides developers with a
+way of describing a neural network so that the Stripe-based PlaidML compiler can
+construct an efficient implementation.
+
+This tutorial is intended to help machine learning practitioners (or anyone with
+a background in software engineering and mathematics) get started using the C++
+Tile EDSL.
+
+## Scope and Warning
+
+This tutorial provides an introduction to the C++ Tile EDSL. It is intended to
+help machine learning practitioners get started writing Tile code as quickly as
+possible, and as such covers core features, not every language detail. This is a
+tutorial, not a spec, and as such will consist of a series of examples, with a
+summary reference section at the end.
+
+This tutorial covers how to use the C++ Tile EDSL, not how Tile code is
+constructed and manipulated by PlaidML. It does not cover the workings of
+PlaidML utilities such as the Tile compiler.
+
+Tile and PlaidML are still in early development and the C++ Tile EDSL is
```

Comment:
This paragraph was written over a year ago and isn't perfectly accurate anymore. Maybe just rewrite this paragraph as
> Tile and PlaidML are still being developed and the APIs discussed here are subject to change.